lib/api/bucketDelete.js

```diff
@@ -42,13 +42,14 @@ function bucketDelete(authInfo, request, log, cb) {
             log.trace('passed checks',
                 { method: 'metadataValidateBucket' });
             return deleteBucket(authInfo, bucketMD, bucketName,
-                authInfo.getCanonicalID(), log, err => {
+                authInfo.getCanonicalID(), log, err, totalPartSizeSum => {
                     if (err) {
                         return cb(err, corsHeaders);
                     }
                     pushMetric('deleteBucket', log, {
                         authInfo,
                         bucket: bucketName,
+                        byteLength: totalPartSizeSum,
```

Comment:
This should be negative, otherwise we would be incrementing storage instead of decrementing it.

Reply:
also v2 only looks at `newByteLength/oldByteLength` to compute `sizeDelta` https://github.com/scality/cloudserver/blob/cd3aac2d8ddbc7ecaa00496b9fc80300873e3fd6/lib/utapi/utilities.js#L262

Reply:
That's what I thought too, but looking at the other APIs that should decrement storageUtilized, they pass the number as a positive integer, like here https://github.com/scality/cloudserver/blob/development/8.2/lib/api/multipartDelete.js#L53 (partSizeSum is positive) and here https://github.com/scality/cloudserver/blob/development/8.2/lib/api/multipartDelete.js#L53

Reply:
Which looking at the utapi functions for those apis, they use the `decr` command. Delete bucket uses the generic push metric call, so would it be too hacky to pass a negative value to the `incr` command? Otherwise, will require some changes in utapi

Reply:
I just looked at that and it seems like its more a bug in the `sizeDelta` bit then. `multiObjectDelete` passes a `isDelete` flag that could be used for v2's `sizeDelta`
https://github.com/scality/cloudserver/blob/cd3aac2d8ddbc7ecaa00496b9fc80300873e3fd6/lib/api/multiObjectDelete.js#L508

Reply:
yeah, I have a feeling that for v1, anything used with the `incrby` command needs to be positive. do you know if that's right?

Reply:
@dora-korpar You are right, I forgot about that part. So the methods issue `incr` or `decr` depending on the api call.